File: lms-frontend/src/student/services/contractService.js
import api from "../../api/axios";

const ADMIN_BASE_URL = "/api/admin/contracts";
const STUDENT_BASE_URL = "/api/student/contracts";

export const uploadContract = async (batchId, file) => {
    const formData = new FormData();
    formData.append("batchId", batchId);
    formData.append("file", file);
    return await api.post(ADMIN_BASE_URL, formData);
};

export const getAllContracts = async () => {
    return await api.get(ADMIN_BASE_URL);
};

export const getContractsByBatch = async (batchId) => {
    return await api.get(`${ADMIN_BASE_URL}/batch/${batchId}`);
};

export const getMyContract = async () => {
    return await api.get(`${STUDENT_BASE_URL}/my-contract`);
};

export const signContract = async (candidateContractId, formData) => {
    return await api.post(
        `${STUDENT_BASE_URL}/${candidateContractId}/sign`,
        formData
    );
};

const contractService = {
    uploadContract,
    getAllContracts,
    getContractsByBatch,
    getMyContract,
    signContract
};

export default contractService;

File: backend/src/main/java/com/lms_erp/student/service/ContractService.java
package com.lms_erp.student.service;

import com.lms_erp.student.dto.ContractResponseDto;
import com.lms_erp.student.entity.Contract;
import org.springframework.web.multipart.MultipartFile;
import java.util.List;

public interface ContractService {

    ContractResponseDto uploadContract(
            Long batchId,
            MultipartFile file,
            Long uploadedByPersonId
    );

    List<ContractResponseDto> getContractsByBatch(Long batchId);

    List<ContractResponseDto> getAllContracts();

    ContractResponseDto getMyContract(Long personId);

    Contract getContractEntity(Long contractId);

    ContractResponseDto signContract(
            Long candidateContractId,
            Long personId,
            String signatureType,
            String signatureData,
            MultipartFile signatureFile
    );

    Contract getContractForStudent(Long contractId, Long personId);
}